src/api/services/adminServiceClient.ts

```diff
@@ -30,16 +30,28 @@ const DEFAULT_SHARING_SETTINGS: Pick<SharingListResponse, 'allowed_sites' | 'all
   allowed_sites: [],
 };
 
-export const getSharingList = async (siteId: number): Promise<SharingListResponse> => {
-  const response = await adminServiceClient.get<SharingListResponse>(
-    `/api/sharing/list/${siteId}`,
-    {
-      validateStatus: (status) => status >= 200 && status < 300,
+export const getSharingList = async (
+  siteId: number,
+  traceId: string
+): Promise<SharingListResponse> => {
+  try {
+    const response = await adminServiceClient.get<SharingListResponse>(
+      `/api/sharing/list/${siteId}`,
+      {
+        validateStatus: (status) => status >= 200 && status < 300,
+      }
+    );
+    return response.data.allowed_sites
+      ? response.data
+      : { ...response.data, ...DEFAULT_SHARING_SETTINGS };
+  } catch (error: unknown) {
+    const { errorLogger } = getLoggers();
+    if (error instanceof AxiosError) {
+      error.message = `Site Id invalid`;
     }
-  );
-  return response.data.allowed_sites
-    ? response.data
-    : { ...response.data, ...DEFAULT_SHARING_SETTINGS };
+    errorLogger.error(`${error}`, traceId);
+    throw error;
```

Comment:
Added a try-catch here because the error in the fallback handler just gives failed with 404 message